content/post/proteus.md

```diff
@@ -0,0 +1,117 @@
+
+**How does it work?**
+
+* Scans all structs that have the comment `//proteus:generate` and generates them as protobuf messages.
+* Scans all the type definitions with the comment `//proteus:generate` and their constant values and transforms them into proper protobuf enumerations.
+* Resolves the types and ignores those which it can't.
+* Converts it to protobuf structures and writes the `.proto` files.
+
+So, imagine you have the following code:
+
+```go
+package models
+
+import "time"
+
+type Model struct {
+        ID        int64
+        CreatedAt time.Time
+}
+
+type User struct {
+        Model
+        Status             Status
+        Username           string
+        Password           string
+        NotCryptedPassword string `proteus:"-"`
+}
+
+type Status int
+
+const (
+        Pending  Status = iota
+        Active
+        Inactive
+)
+```
+
+That code will become the following protobuf file:
+
+```
+syntax = "proto3";
+package models;
+
+import "google/protobuf/timestamp.proto";
+
+message User {
+        int64 id = 1;
+        google.protobuf.Timestamp created_at = 2;
+        models.Status status = 3;
+        string username = 4;
+        string password = 5;
+}
+
+enum Status {
+        PENDING = 0;
+        ACTIVE = 1;
+        INACTIVE = 2;
+}
+```
+
+You can check for a more detailed example in the [examples folder](https://github.com/src-d/proteus/tree/master/example).
+
+You might wonder why we did not use [gengo](https://github.com/kubernetes/gengo) for this task. The reason we did not use it, it is because of the plugin system we want to implement in the future. The plugin system is not implemented yet, we still don't know how it's going to look, but having control of our source code allows us to integrate it in every way we need.
+
+## Next steps
+
+### Marshal and unmarshal
+
+Proteus is far from finished yet. If you haven't noticed, there is one obvious thing missing from this: how do you convert your structs to and from protobuf? We have not yet generated any `Marshal` or `Unmarshal` method.
+
+We are working on that feature in our fork of [`gogo/protobuf`](https://github.com/gogo/protobuf). Instead of generating everything from the generated `.proto` file. We drop all that's generated by `protoc` and only leave the marshal and unmarshal methods.
+
+### gRPC service generation
+
+Right now, you can use the gRPC extension of protobuf to generate the server interface of your services (which have to be defined in your proto file as well). Remember we don't want to have protos as source of truth but Go code.
+
+Proteus will be extended to also generate services with the exported methods and functions.
+
+However, just as it happened with marshal and unmarshal, we still need `protoc` to generate the gRPC server interface. What's the good part of this? Proteus will also generate the gRPC server implementation that implements the interface generated by `protoc`.
+
+### Plugins
+
```

Comment:
You explain that you discarded Gengo just because its plugin system; in that case I'd explain it here (just move that paragraph)